### ISSUE TITLE
generateKeyFrame algorithm leaks pendingKeyFrameTasks

### ISSUE THREAD

**Issue by fippo** [w3c/webrtc-encoded-transform#144]
https://w3c.github.io/webrtc-encoded-transform/#generate-key-frame-algorithm
steps 7 ("Create a pending key frame task...") to 11 are leaking tasks since the algorithm isn't dealing with being called from RTCRtpSender.generateKeyFrame which doesn't have the corresponding pop operation "Remove task from encoder.[[pendingKeyFrameTasks]]."

**Comment by youennf** [comment]
Right, the "remove task from encoder" should run for all encoders, not only for script transformers.
